daphne_worker/src/roles/aggregator.rs

```diff
@@ -465,31 +464,28 @@ impl<'srv> DapAggregator<DaphneWorkerAuth> for DaphneWorker<'srv> {
         .flatten()
         .collect::<HashSet<ReportId>>();
 
```

Comment:
Suppose there are two DO calls made: the first detects no replays and thus marks all reports as aggregated, but another detects replays and so returns the subset that were replayed. What happens?

If i understand the code correctly, `replayed` will be non-empty, so we'll return `Ok(Some(replayed))`. When we retry, the reports that were marked aggregated on the first attempt will be considred replays on the second try. Do I have that right?

If so, I think we need to separate `MARK_AGGREGATED` into two phases. In the first phase we check for replays; in the second we either commit all the state changes or none of them.

Reply:
No, mark aggregated is an all or nothing call now. Hence that complex type in the `ReportsProcessed` durable object. Either no reports are fresh or nothing is considered aggregated. This is a change from the old behavior of this call

Reply:
Per discussion, let's resolve this for now by just leaving a TODO noting the bug. (Since it exists before this PR.)